Plugins/GitUIPluginInterfaces/ObjectId.cs

```diff
@@ -43,13 +43,27 @@ public static ObjectId Random()
             return new ObjectId(
                 unchecked((ulong)_random.NextInt64()),
                 unchecked((ulong)_random.NextInt64()),
-                unchecked((uint)_random.Next()));
+                unchecked((uint)_random.Next()),
+                0,
+                0,
+                false);
         }
 
+        /// <summary>
+        /// Checks if a char buffer with the given length can be a valid hash.
+        /// </summary>
+        /// <param name="length">The length of the buffer.</param>
+        /// <returns>True if the buffer can be a valid hash; otherwise false. It must still be checked if the buffer contains a valid hash.</returns>
+        public static bool IsValidLength(int length) => length == Sha1CharCount || length == Sha256CharCount;
```

Comment:
Perhaps `IsValidCharacterLength`, to be more specific about what is being counted.